engine/classes/Elgg/Views/View.php

```diff
@@ -0,0 +1,211 @@
+<?php
+namespace Elgg\Views;
+
+use Elgg\Filesystem\File;
+use Elgg\DeprecationWrapper;
+
+/**
+ * @access private
+ */
+class View {
+	/** @var bool */
+	private $isCacheable = false;
+	
+	/** @var File[] */
+	private $locations = array();
+	
+	/** @var View[] */
+	private $prepends = array();
+	
+	/** @var View[] */
+	private $appends = array();
+
+	/**
+	 * Whether the view has been registered/extended.
+	 * 
+	 * @param string  $viewtype
+	 * @param boolean $recurse
+	 * 
+	 * @return boolean
+	 */
+	public function exists(Viewtype $viewtype, $recurse = true) {
+		$location = $this->getLocation($viewtype);
+
+		if (!empty($location)) {
+			return true;
+		}
+
+		// If we got here then check whether this exists as an extension
+		if ($recurse) {
+			$extensions = array_merge($this->prepends, $this->appends);
+			foreach ($extensions as $extension) {
+				// do not recursively check to stay away from infinite loops
+				if ($extension->exists($viewtype, false)) {
+					return true;
+				}
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * @param View $extension
+	 * @param int  $priority
+	 * 
+	 * @access private
+	 */
+	public function prepend(View $extension, $priority) {
```

Comment:
It does, but I map it to this more explicit API.

I dunno maybe that's a bad idea?